spanner/tests/system.py

```diff
@@ -228,16 +247,49 @@ def test_update_database_ddl(self):
 
         self.assertEqual(len(temp_db.ddl_statements), len(DDL_STATEMENTS))
 
+    def test_db_batch_insert_then_db_snapshot_read_and_db_read(self):
+        from google.cloud.spanner import KeySet
+        keyset = KeySet(all_=True)
 
-class TestSessionAPI(unittest.TestCase):
-    TABLE = 'contacts'
-    COLUMNS = ('contact_id', 'first_name', 'last_name', 'email')
-    ROW_DATA = (
-        (1, u'Phred', u'Phlyntstone', u'phred@example.com'),
-        (2, u'Bharney', u'Rhubble', u'bharney@example.com'),
-        (3, u'Wylma', u'Phlyntstone', u'wylma@example.com'),
-    )
-    SQL = 'SELECT * FROM contacts ORDER BY contact_id'
+        retry = RetryInstanceState(_has_all_ddl)
+        retry(self._db.reload)()
+
+        with self._db.batch() as batch:
+            batch.delete(self.TABLE, keyset)
```

Comment:
Races about previous uses of the same table (we can't create new tables cheaply on the fly).